0002-crypto-conditions/0002-crypto-conditions.md

```diff
@@ -114,312 +114,288 @@ This specification is a part of the [Interledger Protocol](https://interledger.o
 
-## PREIMAGE {#preimage-feature-suite}
-The PREIMAGE feature suite provides conditions that use a preimage as a one-time signature. This feature suite is assigned the feature bit 2^1 = 0x02.
+Each bit represents a type. The list of known types is the IANA-maintained [Crypto-Condition Type Registry](#crypto-conditions-type-registry) .
 
-The fingerprint of a preimage condition is the hash of an arbitrary value. The payload of a preimage fulfillment is the hashed arbitrary value before hashing, also known as the preimage. Conditions that use this preimage MUST also rely on a cryptographically secure hashing algorithm. Since cryptographically secure hashing functions are preimage-resistant, only the original creator of a preimage condition can produce the preimage, as long as it contains a large amount of random entropy.
+The presence of one of more sub-crypto-conditions of a specific type is indicated by setting the numbered bit corresponding to the type number of that type. Bits are numbered from the least significant bit of the first byte to the most significant bit of the last byte.
 
-## PREFIX {#prefix-feature-suite}
-The PREFIX feature suite provides conditions that prepend a fixed message to a subcondition. This feature suite is assigned the feature bit 2^2 = 0x04.
+#### Examples
 
-A prefix condition prepends the message to be validated with a constant string before passing it on to the subcondition's validation function. <!-- TODO: better explanation of why -->
+The following bitmask indicates the presence of the subtypes PREIMAGE-SHA-256 (type number 0) and RSA-SHA-256 (type number 3)
 
-## THRESHOLD {#threshold-feature-suite}
-The THRESHOLD feature suite provides conditions that have several weighted subconditions and a threshold number. This feature suite is assigned the feature bit 2^3 = 0x08.
+    Hex
+    1    2    3    4    
+    0x09 0x00 0x00 0x00
+    
+    Binary
+    1         2         3         4          
+    0000 1001 0000 0000 0000 0000 0000 0000  
+    
+The following bitmask indicates the presence of the subtype ED25519 (type number 4)
 
-Threshold conditions provide flexible multi-signing, such as requiring "M-of-N" subconditions be fulfilled. Subconditions can also be weighted so that one subcondition can count multiple times towards meeting the threshold.
+    Hex
+    1    2    3    4    
+    0x10 0x00 0x00 0x00 
+    
+    Binary
+    1         2         3         4           
+    0001 0000 0000 0000 0000 0000 0000 0000  
+    
+The following bitmask indicates the presence of an unknown subtype with type number 10.
 
-## RSA-PSS {#rsa-pss-feature-suite}
-The RSA-PSS feature suite provides the RSS-PSA signature algorithm. RSA-PSS is a signature algorithm based on the RSA cryptosystem, which relates to the problem of factoring the product of two large prime numbers. This feature suite is assigned the feature bit 2^4 = 0x10.
+    Hex
+    1    2    3    4    
+    0x00 0x04 0x00 0x00
```

Comment:
Second byte of the bitmap is 0x04. 

Bit 2 of the second byte is set indicating that there is a sub condition used within this compound condition that has the type ID 10. i.e. If you see this then reject it because nobody knows what type 10 is 😄